cluster/engine.go

```diff
@@ -192,6 +196,27 @@ func (e *Engine) Status() string {
 	return "Unhealthy"
 }
 
+// IncFailureCount increases engine's failure count, and set engine as unhealthy if threshold is crossed
+func (e *Engine) IncFailureCount() {
```

Comment:
I think this should be private (`incFailureCount`?).

No-one outside should be calling this - I think internally we should call it whenever one of the dockerclient calls fails.

You might do a helper, like:

```go
foo, err := e.client.ListContainers()
e.checkTransportError(err)
[... rest of the code as usual ...]

func (e *Engine) checkTransportErr(err error) {
    if (check if it's a transport error from docker client) {
        e.increaseFailureCount()
    } else {
        e.resetFailureCount()
   }
}
```

Or something like that.

We would call `checkTransportError` **every** time we use `dockerclient`, meaning that every call to the engine would contribute to the failure count.

Thoughts?